### ISSUE TITLE
servicesSubnet is hardcoded to 10.3.0.0

### ISSUE THREAD

**Issue by vrutkovs** [openshift/cluster-kube-apiserver-operator#143]
`servicesSubnet` is set to 10.3.0.0 in master's kube API server - https://github.com/openshift/cluster-kube-apiserver-operator/blob/4f197523d36c70e06e29fc758452e9638fd20d53/bindata/v3.11.0/kube-apiserver/defaultconfig.yaml#L89
Installer has `serviceCIDR` set to `172.30.0.0/16` so most namespaces have `Cluster IP 172.30.220.6 is not within the service CIDR 10.3.0.0/16; please recreate service` event

**Comment by wking** [comment]
The installer default changed in openshift/installer@c6c9d832e94d0c677f3d8e6170e4306dd00df23a (openshift/installer#712).  Ideally operators should be pulling this value from [the network config][1].

[1]: https://github.com/openshift/installer/blob/v0.4.0/pkg/asset/manifests/network.go#L107-L118

**Comment by wking** [comment]
Example of pulling thihis from the network config:

```console
$ oc get -o jsonpath="{.spec.serviceNetwork}{'\n'}" networkconfig default
172.30.0.0/16
```

**Comment by smarterclayton** [comment]
Operators are not manually setting this.  Services created by the CVO are getting assigned the 172.x block, and then it looks like we change what the value is.  The CVO service for instance has no hardcoded IP and is getting a 172.

**Comment by vrutkovs** [comment]
This is breaking openshift API server on GCP setup:
```
E1127 13:27:12.158406       1 controller.go:111] loading OpenAPI spec for "v1.image.openshift.io" failed with: failed to retrieve openAPI spec, http error: ResponseCode: 503, Body: service unavailable

E1127 13:27:50.036693       1 available_controller.go:327] v1.template.openshift.io failed with: Get https://172.30.222.191:443: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```
```
# oc get svc -n openshift-apiserver
NAME      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
api       ClusterIP   172.30.222.191   <none>        443/TCP   47m
```
```
# oc get events -n openshift-apiserver
1m          43m          16        api.156afbe83ec48845               Service                                            Warning   ClusterIPOutOfRange      ipallocator-repair-controller       Cluster IP 172.30.222.191 is not within the service CIDR 10.3.0.0/16; please recreate service
52s         42m          16        api.156afbed986f0063               Service                                            Warning   ClusterIPOutOfRange      ipallocator-repair-controller       Cluster IP 172.30.222.191 is not within the service CIDR 10.3.0.0/16; please recreate service
11s         45m          17        api.156afbcd22e6d3bb               Service                                            Warning   ClusterIPOutOfRange      ipallocator-repair-controller       Cluster IP 172.30.222.191 is not within the service CIDR 10.3.0.0/16; please recreate service
```

**Comment by sttts** [comment]
I believe this is also due to a race between our config observers seeing a valid and updated config and the operator installing phase 2 pods blindly to the master. We have to make sure that the config is valid and complete before an install is tried.

Working on this.

**Comment by vrutkovs** [comment]
Fixed in #146

**Comment by jmrodri** [comment]
Is there a workaround for this? using the openshift-installer from master and apiserver just continues to get restarted.

**Comment by djwhatle** [comment]
Also interested in a workaround!

**Comment by sttts** [comment]
@vrutkovs you reopened this. Is this still a problem? (I checked in a local cluster and all services had correct IPs). If not, please close.

**Comment by vrutkovs** [comment]
Fixed by #147, `servicesSubnet` is correct and no services are created with `10.3....`
